src/Lumina/Models/Materials/Material.cs

```diff
@@ -0,0 +1,230 @@
+            VariantId = variantId;
+        }
+
+        /// <summary>
+        /// Creates a new Material instance using the provided MtrlFile.
+        /// Variant ID is inferred from the provided MtrlFile's path.
+        /// </summary>
+        /// <param name="file">The MtrlFile to back this Material.</param>
+        public Material( MtrlFile file )
+        {
+            File = file;
+            VariantId = GetVariantIdFromPath( file.FilePath );
+            BuildMaterial();
+        }
+
+        /// <summary>
+        /// Creates a new Material instance using the provided path and
+        /// reference to game data. The Material will be built and then
+        /// updated with game data.
+        /// </summary>
+        /// <param name="data">A reference to game data access.</param>
+        /// <param name="path">The path, relative or absolute, to this Material.</param>
+        /// <param name="variantId">The variant ID for this material. Default is 1.</param>
+        public Material( GameData data, string path, int variantId = 1 )
+        {
+            MaterialPath = path;
+            BuildMaterial();
+            Update( data );
+        }
+
+        /// <summary>
+        /// Update this Material using game data. If instantiated without a MtrlFile or
+        /// GameData, this method will retrieve the MtrlFile referenced by this Material and use
+        /// that file to update local fields. This method is not guaranteed to load the MtrlFile.
+        /// </summary>
+        /// <param name="data">A reference to game data access.</param>
+        /// <returns>The existing Material instance, for method chaining.</returns>
+        public Material Update( GameData data )
+        {
+            if( MaterialPath.StartsWith( "/" ) )
+            {
+                ResolvedPath = ResolveRelativeMaterialPath( MaterialPath, VariantId );
+                File = data.GetFile< MtrlFile >( ResolvedPath );
+            }
+            else
+            {
+                File = data.GetFile< MtrlFile >( MaterialPath );
+            }
+
+            if( File != null )
+                BuildMaterial();
+
+            return this;
+        }
+
+        /// <summary>
+        /// Resolves a relative material path in the form <c>/mt_c0101e0001_top_a.mtrl</c>
+        /// into its full path, <c>chara/equipment/e0001/material/v{variantId}/mt_c0101e0001_top_a.mtrl</c>.
+        /// <br/>This method will successfully resolve all known relative material paths.
+        /// </summary>
+        /// <param name="relativePath">The relative path of the provided material.</param>
+        /// <param name="variantId">The variant to use in material resolution.</param>
+        /// <returns>The resolved, absolute path to the requested material, or null if unsuccessful.</returns>
+        public static string ResolveRelativeMaterialPath( string relativePath, int variantId )
+        {
+            var id1 = char.Parse( relativePath.Substring( 4, 1 ) );
```

Comment:
you don't have to char.parse this, you can just substring(...)[0], or just relativePath[4]

Reply:
fixed